### PR TITLE
[DJFS-111] Develop a customer details list view page

### PR DESCRIPTION
**Pull request by Cynthia-Berry**
## Task: `[DJFS-111] Develop a customer details list view page`
- ``

### CODE REVIEW

File: submissions/cynthiaNwakaeme/php-database/cookies.php [hunk 1]
```diff
@@ -1,55 +1,68 @@
...
 
-try{
-    /* ACCESSING E-COMMERCE DATABASE AND QUERYING THE DATABASE*/
-    $use_db = "USE taries_empire";
-    $sql = "SELECT * FROM customers";
-    $pdo->exec($use_db);
-    $stmt = $pdo->query($sql);
-    
+class Customer{
+    public function dbQuery(){
```
Comment (adaugochi) [inline submissions/cynthiaNwakaeme/php-database/cookies.php:6]: This function needs to return something

**Review by adaugochi — CHANGES_REQUESTED** [review]

File: submissions/cynthiaNwakaeme/php-database/cookies.php [hunk 2]
```diff
@@ -1,55 +1,69 @@
...
+            $pdo->exec($use_db);
+            $stmt = $pdo->query($sql);
+            
+            /* GENERATING COOKIES
+            1. result is initalized to get all records from database
+            2. This result is serialize to change the array/object to simple string/text file which is required to create cookies.
+             */
+             $result = $stmt->fetchAll();
+            setcookie("customer", serialize($result), time() + (86400 * 30), "/", "", false); //60sec *60mins *24hours/day => 86400 = 1 day   
+            return($result);
```
Comment (adaugochi) [inline submissions/cynthiaNwakaeme/php-database/cookies.php:22]: It is `return $result` not `return($result)`

**Review by adaugochi — CHANGES_REQUESTED** [review]

File: submissions/cynthiaNwakaeme/php-database/cookies.php (hunk 2 above)
Comment (Cynthia-Berry) [reply submissions/cynthiaNwakaeme/php-database/cookies.php:22]: Thank You ma.

**Review by adaugochi — APPROVED** [review]

File: submissions/cynthiaNwakaeme/php-database/cookies.php [hunk 3]
```diff
@@ -1,55 +1,69 @@
...
+            echo"<th> Full Name </th>";
+            echo"<th> Email Address </th>";
+            echo"<th> Created At </th>";
+            echo"<th> Actions </th>";
+        echo"</tr>";
+    // Loop through to access records (Objects in an array i.e datatype <=> JSON formatt)
+    foreach($output as $row){
+        echo"<tr>";
+            echo "<td>" . $row['id'] ."</td>";
+            echo "<td>" . $row['full_name'] ."</td>";
```
Comment (adaugochi) [inline submissions/cynthiaNwakaeme/php-database/cookies.php:55]: Yout customers table does not have any column called `full_name` 
image

Get the result for Full name from `first_name` and `last_name`

**Review by adaugochi — DISMISSED** [review]

File: submissions/cynthiaNwakaeme/php-database/cookies.php (hunk 3 above)
Comment (Cynthia-Berry) [reply submissions/cynthiaNwakaeme/php-database/cookies.php:55]: Hello @adaugochi , I am so sorry i am reverting late. I only just figured out that I had to use the existing database. I actually created another for this task.

Thank you for pointing me to that. Corrections has been effected. Kindly review.

Comment (Cynthia-Berry) [reply submissions/cynthiaNwakaeme/php-database/cookies.php]: Also, I noticed there is no created_at in the customers table so I used the one in the orders table

**Review by adaugochi — APPROVED** [review]

File: submissions/cynthiaNwakaeme/php-database/cookies.php [hunk 4]
```diff
@@ -3,53 +3,60 @@
...
+    }
+}
 
-    /* GENERATING COOKIES
-    1. result is initalized to get all records from database
-    2. This result is serialize to change the array/object to simple string/text file which is required to create cookies.
-     */
-     $result = $stmt->fetchAll();
-    setcookie("customer", serialize($result), time() + (86400 * 30), "/", "", false); //60sec *60mins *24hours/day => 86400 = 1 day
+$try = new Customer();
```
Comment (adaugochi) [inline submissions/cynthiaNwakaeme/php-database/cookies.php:30]: use `$customer` instead of `$try`

**Review by adaugochi — CHANGES_REQUESTED** [review]

File: submissions/cynthiaNwakaeme/php-database/cookies.php (hunk 4 above)
Comment (Cynthia-Berry) [reply submissions/cynthiaNwakaeme/php-database/cookies.php]: Done, Thank you ma.

**Pull request merged**